src/protocols/utils/dist.c

```diff
@@ -71,14 +71,13 @@ int nn_dist_send (struct nn_dist *self, struct nn_msg *msg,
     struct nn_dist_data *data;
     struct nn_msg copy;
 
-    /*  TODO: We can optimise for the case when there's only one outbound
+    /*  TODO: We can optimize for the case when there's only one outbound
         pipe here. No message copying is needed in such case. */
 
     /*  In the specific case when there are no outbound pipes. There's nowhere
-        to send the message to. Deallocate it. */
+        to send the message to. */
     if (nn_slow (self->count) == 0) {
-        nn_msg_term (msg);
-        return 0;
+        return -EAGAIN;
```

Comment:
I will check, but I think it is related. Windows would silently drop messages otherwise, and we need a way to tell when it wasn't sent due to a socket with a full buffer.